client/service/src/metrics.rs

```diff
@@ -0,0 +1,379 @@
+			metrics.cpu_usage_percentage.set(process_info.cpu_usage as f64);
+			// `sysinfo::Process::memory` returns memory usage in KiB and not bytes.
+			metrics.memory_usage_bytes.set(process_info.memory * 1024);
+
+			if let Some(threads) = process_info.threads {
+				metrics.threads.set(threads);
+			}
+
+			if let Some(fd_info) = process_info.open_fd {
+				metrics.open_files.with_label_values(&["paths"]).set(fd_info.paths);
+				metrics.open_files.with_label_values(&["mem"]).set(fd_info.mem);
+				metrics.open_files.with_label_values(&["sockets"]).set(fd_info.sockets);
+				metrics.open_files.with_label_values(&["net"]).set(fd_info.net);
+				metrics.open_files.with_label_values(&["pipe"]).set(fd_info.pipes);
+				metrics.open_files.with_label_values(&["anon_inode"]).set(fd_info.anon_inode);
+				metrics.open_files.with_label_values(&["other"]).set(fd_info.other);
+			}
+
+			let load = self.system.get_load_average();
+			metrics.load_avg.with_label_values(&["1min"]).set(load.one);
+			metrics.load_avg.with_label_values(&["5min"]).set(load.five);
+			metrics.load_avg.with_label_values(&["15min"]).set(load.fifteen);
+
+			metrics.network_per_sec_bytes.with_label_values(&["download"]).set(net_status.average_download_per_sec);
+			metrics.network_per_sec_bytes.with_label_values(&["upload"]).set(net_status.average_upload_per_sec);
+
+			metrics.chain_info.with_label_values(&["finalized"]).set(finalized_number);
+			metrics.chain_info.with_label_values(&["best"]).set(best_number);
+			if let Ok(leaves) = u64::try_from(info.chain.number_leaves) {
+				metrics.chain_info.with_label_values(&["leaves"]).set(leaves);
+			}
+
+			metrics.ready_transactions_number.set(txpool_status.ready as u64);
+
+			if let Some(best_seen_block) = best_seen_block {
+				metrics.chain_info.with_label_values(&["sync_target"]).set(best_seen_block);
+			}
+
+			if let Some(info) = info.usage.as_ref() {
+				metrics.database_cache.set(info.memory.database_cache.as_bytes() as u64);
+				metrics.state_cache.set(info.memory.state_cache.as_bytes() as u64);
+
+				metrics.state_db.with_label_values(&["non_canonical"]).set(info.memory.state_db.non_canonical.as_bytes() as u64);
+				if let Some(pruning) = info.memory.state_db.pruning {
+					metrics.state_db.with_label_values(&["pruning"]).set(pruning.as_bytes() as u64);
+				}
+				metrics.state_db.with_label_values(&["pinned"]).set(info.memory.state_db.pinned.as_bytes() as u64);
+			}
+
+			if let Some(conns) = self.connections_info() {
+				metrics.netstat.with_label_values(&["listen"]).set(conns.listen);
+				metrics.netstat.with_label_values(&["established"]).set(conns.established);
+				metrics.netstat.with_label_values(&["starting"]).set(conns.starting);
+				metrics.netstat.with_label_values(&["closing"]).set(conns.closing);
+				metrics.netstat.with_label_values(&["closed"]).set(conns.closed);
+				metrics.netstat.with_label_values(&["other"]).set(conns.other);
+			}
+		}
+	}
+}
```

Comment:
Missing newline at EOF

```suggestion
}

```